src/sentry/features/manager.py

```diff
@@ -66,8 +68,37 @@ def has(self, name, *args, **kwargs):
 
         >>> FeatureManager.has('organizations:feature', organization, actor=request.user)
         """
+        return self.build_checker().has(name, *args, **kwargs)
+
+    def build_checker(self):
+        """
+        Set up for a delayed execution of ``has``.
+
+        Successive calls to the returned checker have better performance than
+        repeatedly calling ``FeatureManager.has``, because it retains the set
+        of ``feature.handler.FeatureHandler`` objects. An instance of the
+        checker should be kept only in a short-lived context, because any
+        dynamic changes to plugins' feature handlers are not reflected in its
+        behavior.
+        """
+        return FeatureChecker(self)
+
+
+class FeatureChecker(object):
+    def __init__(self, manager):
+        from sentry.plugins.base import plugins
+
+        self.manager = manager
+
+        handlers_per_plugin = [
+            safe_execute(plugin.get_feature_hooks, _with_transaction=False) or ()
```

Comment:
@RyanSkonnord I am not sure what the initial intention of `get_feature_hooks` but I can assure you don't need to include this anymore. No plugins have `get_feature_hooks` subclassed to return anything but an empty array. Furthermore, the only plugins which have feature restrictions are not even V2 plugins so that are yielded from `plugins.all(version=2)`. So I think you should just remove this part of the code since `handlers_per_plugin` is always going to be an empty array.

Reply:
That's important information! That means we can skip looking at plugins entirely and, by extension, don't need this entire "checker" abstraction.

To be perfectly clear, does this mean you would be comfortable nuking the entire `get_feature_hooks` method from the base Plugin class? Because removing the reference to it here would feel safer to me if it didn't exist at all.

Reply:
@RyanSkonnord Yes, I think nuking the method is the way to go. If we ever need to actually enforce feature gating on plugins (which is unlikely at this point), we can find a different/better way.

Reply:
Just to be clear here, I missed the fact we register `GetSentryFeaturesPlugin` as a plugin. That plugin isn't real plugin but it's used to determine what features a plan maps to. We don't need any other plugins for feature gating but we do need that fake plugin, or someway of replicating that logic.